File: Visual Studio Product Team/Official Visual Studio 2010 Samples for C# 4.0/46872-Official Visual Studio 2010 Samples for C# 4.0/ADO.NET Entity Framework/VB/Common/IEmployeeRepository.vb
' Copyright © Microsoft Corporation.  All Rights Reserved.
' This code released under the terms of the 
' Microsoft Public License (MS-PL, http://opensource.org/licenses/ms-pl.html.)



Imports Microsoft.VisualBasic
Imports System.Collections.Generic
Imports EmployeeTracker.Model

Namespace EmployeeTracker.Model.Interfaces

    ''' <summary>
    ''' Repository for retrieving employee data
    ''' </summary>
    Public Interface IEmployeeRepository
        ''' <summary>
        ''' All employees for the company
        ''' </summary>
        ''' <returns>Enumerable of all employees</returns>  
        Function GetAllEmployees() As IEnumerable(Of Employee)

        ''' <summary>
        ''' Gets the longest serving employees
        ''' </summary>
        ''' <param name="quantity">The number of employees to return</param>
        ''' <returns>Enumerable of the longest serving employees</returns>
        Function GetLongestServingEmployees(ByVal quantity As Integer) As IEnumerable(Of Employee)
    End Interface
End Namespace

File: Visual Studio Product Team/Official Visual Studio 2010 Samples for C# 4.0/46872-Official Visual Studio 2010 Samples for C# 4.0/ADO.NET Entity Framework/C#/Common/IEmployeeRepository.cs
// Copyright © Microsoft Corporation.  All Rights Reserved.
// This code released under the terms of the 
// Microsoft Public License (MS-PL, http://opensource.org/licenses/ms-pl.html.)


namespace EmployeeTracker.Model.Interfaces
{
    using System.Collections.Generic;
    using EmployeeTracker.Model;

    /// <summary>
    /// Repository for retrieving employee data
    /// </summary>
    public interface IEmployeeRepository
    {
        /// <summary>
        /// All employees for the company
        /// </summary>
        /// <returns>Enumerable of all employees</returns>  
        IEnumerable<Employee> GetAllEmployees();

        /// <summary>
        /// Gets the longest serving employees
        /// </summary>
        /// <param name="quantity">The number of employees to return</param>
        /// <returns>Enumerable of the longest serving employees</returns>
        IEnumerable<Employee> GetLongestServingEmployees(int quantity);
    }
}
